src/components/TodoItem.tsx

```diff
@@ -0,0 +1,167 @@
+import classNames from 'classnames';
+import { useState, useEffect, useRef } from 'react';
+import { Todo } from '../types/Todo';
+
+interface Item {
+  todo: Todo,
+  loaderId: number[],
+  removeTodo: (todoId: number) => Promise<unknown>,
+  updateTodo: (todoId: number) => Promise<unknown>,
+  updateTodoTitle: (todoId: number) => Promise<unknown>,
+  switchTodoTitle: (newTitle: string) => void,
+  newTodoTitle: string,
+}
+
+export const TodoItem: React.FC<Item> = ({
+  todo,
+  loaderId,
+  removeTodo,
+  updateTodo,
+  updateTodoTitle,
+  switchTodoTitle,
+  newTodoTitle,
+}) => {
+  const [loadingTodo, setLoadingTodo] = useState(false);
+  const [editing, setEditing] = useState(false);
+
+  const selectedField = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (editing && selectedField.current) {
+      selectedField.current.focus();
+    }
+  }, [editing]);
+
+  function enterEdit() {
+    switchTodoTitle(todo.title);
+    setEditing(true);
+  }
+
+  function clearEdit() {
+    switchTodoTitle('');
+    setEditing(false);
+  }
+
+  const handleToggle = () => {
```

Comment:
better to write in one style
use `function name() {` to `const name = () => {` (second approach is better)